### ISSUE TITLE
ngh doesn't seem to recognize GH_TOKEN

### ISSUE THREAD

**Issue by amivit** [angular-buch/angular-cli-ghpages#20]
My personal [job log](https://travis-ci.org/amivit/RSMuseum-ng/jobs/235940061#L275-L284) from a failed deployment.
My personal [.travis.yml](https://github.com/amivit/RSMuseum-ng/blob/b29bb9391ffe4bdcadb6e018185f16d90ef2dd5d/.travis.yml) from this specific project

I'm pretty sure I've setup the [environment variable](http://i.imgur.com/Eps3MvN.png) on Travis correctly, including the token permissions. Am I missing something silly, or is there a potential bug?

**Comment by amivit** [comment]
I'm not too bright - the error was right there in the job log. Git needs to be properly configured. My bad!

**Comment by amivit** [comment]
Oh, it [still fails](https://play.google.com/store/apps/details?id=io.ionic.amivit&hl=en) even with [git configured](https://play.google.com/store/apps/details?id=io.ionic.amivit&hl=en)

**Comment by amivit** [comment]
Oh, it [still fails](https://play.google.com/store/apps/details?id=io.ionic.amivit&hl=en) even with [git configured](https://play.google.com/store/apps/details?id=io.ionic.amivit&hl=en)

**Comment by JohannesHoppe** [comment]
The message in the log is pretty clear:
>   message: 'remote: Invalid username or password.\nfatal: Authentication failed for \'https://github.com/amivit/RSMuseum-ng.git/\'\n',

The configured username and password will make your git history pretty. They do not have any influence to the authentication to github. 

Please see this:
![image](https://cloud.githubusercontent.com/assets/640639/26482875/0e992c62-41ea-11e7-9e20-5ff740dac1f9.png)
Travis has no rights to WRITE to your github repos (that's good!).

It will be a bit complicated to change the global git configuration to use another account. I never tried that for travis. The easiest solution is the following:

1. Please create a GH_TOKEN as described here:
https://github.com/blog/1509-personal-api-tokens
![image](https://cloud.githubusercontent.com/assets/640639/26482962/a18768e0-41ea-11e7-8030-79d6fb4c8328.png)

2. Set the Travis Environment variable `GH_TOKEN`.
3. Execute `ngh --repo=https://GH_TOKEN@github.com/amivit/RSMuseum-ng.git --name="Name just for history" --email=mail_just_for_the_history@example.org`

**Comment by amivit** [comment]
Ohhh, I believe I misunderstood [this](https://github.com/angular-buch/angular-cli-ghpages#extra) and though ngh would automatically recognize the GH_TOKEN env var (that IS present). But I actually have to include "GH_TOKEN" within the --repo option. I didn't do that and was just executing ngh. Let me try again.

**Comment by JohannesHoppe** [comment]
Yes, it recognizes the GH_TOKEN env var and replaces the same string in the URL.  No magic here! ;-)

**Comment by amivit** [comment]
Yeah it works perfect now, thanks!
